internal/control_plane/server/services.go

```diff
@@ -19,8 +21,10 @@ func NewProcServiceServer(exec execution.Execution) protobuf.ClientCPServicesSer
 }
 
 func (s *clientCPServicesServer) PostMetadata(ctx context.Context, request *protobuf.RequestToPostMetadata) (*protobuf.MetadataName, error) {
+	uid := id_generator.NextID()
+	ctx = context.WithValue(ctx, "uid", uid)
 	name, err := s.procExec.SaveMetadataToDb(ctx, request.Metadata)
-	logger.Error(err, "Job Create Request Received - Posting Metadata to etcd")
+	logger.Error(err, fmt.Sprintf("%v Job Create Request Received - Posting Metadata to etcd", uid))
```

Comment:
no error checking? please check the error and only log/return if it's only `err != nil`. this will always log the error even the error is nil.

Reply:
In Zerolog.go, we are using .Err(err).Msg(msg) with Logger, that logs error if error is nil, otherwise logs the message as info.

Reply:
May be we can rename the logger.Error to something like logger.ErrorCheck , will that make more sense @faruqisan ?

Reply:
that's not a good design. please don't create magic like that, the practice is to check the error as near as possible to the cause. please use log.Info to do log the info and `log.Error` to log the error don't log the info in `log.Error` if error is nil.

Reply:
okay, thanks for the suggestion. I guess we need to implement this suggestions at a lot more places, in cli logging too. Will do that!

Reply:
Created issue #40 for same, as a lot of files outside the scope of this PR will be effected by this design change.